ikarus/finiteelements/mechanics/materials/hyperelastic/deviatoric/blatzko.hh

```diff
@@ -0,0 +1,132 @@
+#include <ikarus/finiteelements/mechanics/materials/materialhelpers.hh>
+#include <ikarus/finiteelements/mechanics/materials/tags.hh>
+#include <ikarus/utils/tensorutils.hh>
+
+namespace Ikarus::Materials {
+
+/**
+ * \brief Implementation of the Blatz-Ko material model.
+ * \ingroup materials
+ *
+ * \details The energy is computed as
+ * \f[ \hat{\Psi}(\la_1, \la_2, \la_3) = \frac{\mu}{2}(
+ \frac{1}{\la_1^2} + \frac{1}{\la_2^2} + \frac{1}{\la_3^2}
+ + 2\la_1 \la_2 \la_3 - 5). \f]
+ *
+ * \remark A special Blatz-Ko material model is implemented here. It assumes material parameters which corresponds to a
+ * Poisson's ratio of 0.25.
+ *
+ * \tparam ST The underlying scalar type.
+ */
+template <typename ST>
+struct BlatzKoT
+{
+  using ScalarType         = ST;
+  using PrincipalStretches = Eigen::Vector<ScalarType, 3>;
+
+  static constexpr int dim         = 3;
+  static constexpr auto stretchTag = PrincipalStretchTag::total;
+
+  using FirstDerivative  = Eigen::Vector<ScalarType, dim>;
+  using SecondDerivative = Eigen::Matrix<ScalarType, dim, dim>;
+
+  using MaterialParameters = double;
+
+  [[nodiscard]] constexpr static std::string name() noexcept { return "BlatzKo"; }
+
+  /**
+   * \brief Constructor for BlatzKoT.
+   * \param mu The shear modulus.
+   */
+  explicit BlatzKoT(MaterialParameters mu)
+      : mu_{mu} {}
+
+  /**
+   * \brief Returns the material parameters stored in the material
+   */
+  MaterialParameters materialParametersImpl() const { return mu_; }
+
+  /**
+   * \brief Computes the stored energy in the BlatzKo material model.
+   *
+   * \param lambda principal stretches
+   * \return ScalarType
+   */
+  ScalarType storedEnergyImpl(const PrincipalStretches& lambda) const {
+    return mu_ / 2 *
+           (1 / pow(lambda[0], 2) + 1 / pow(lambda[1], 2) + 1 / pow(lambda[2], 2) +
+            2 * lambda[0] * lambda[1] * lambda[2] - 5);
+  }
+
+  /**
+   * \brief Computes the first derivative of the stored energy function w.r.t. the total principal stretches.
+   *
+   * \param lambda principal stretches
+   * \return ScalarType
+   */
+  FirstDerivative firstDerivativeImpl(const PrincipalStretches& lambda) const {
+    auto dWdLambda     = FirstDerivative::Zero().eval();
+    const ScalarType J = Impl::determinantFromPrincipalValues<ScalarType>(lambda);
+
+    for (auto k : dimensionRange())
+      dWdLambda[k] = mu_ * (-1.0 / pow(lambda[k], 3) + (J / lambda[k]));
+
+    return dWdLambda;
+  }
+
+  /**
+   * \brief Computes the second derivatives of the stored energy function w.r.t. the total principal stretches.
+   *
+   * \param lambda principal stretches
+   * \return ScalarType
+   */
+  SecondDerivative secondDerivativeImpl(const PrincipalStretches& lambda) const {
+    auto dS            = SecondDerivative::Zero().eval();
+    const ScalarType J = Impl::determinantFromPrincipalValues<ScalarType>(lambda);
+
+    for (const auto i : dimensionRange())
```

Comment:
This can be simplified with 

```cpp
       Eigen::Matrix3d dS2;
   auto lam = lambda.array();
    dS2 = J /(lambda * lambda.transpose()).array();
    dS2.diagonal()= lam.square().inverse()*  (lam.square().inverse()-J)+ (3.0/lam.pow(4));
    dS2.array() *= mu_;
```